modin/data_management/query_compiler/pandas_query_compiler.py

```diff
@@ -409,11 +426,14 @@ def reindex_partition(df):
             # The challenge with avoiding reindexing is that we need to make sure that
             # the internal indices line up (i.e. if a drop or a select was just
             # performed, the internal indices may not match).
-            if i != 0:
+            if i != 0 or (left_old_idx.equals(joined_index) and not force_repartition):
```

Comment:
```suggestion
            if (i != 0 or (left_old_idx.equals(joined_index)) and not force_repartition):
```
parenthesis for readability